rust/lance-encoding/src/compression.rs

```diff
@@ -475,7 +478,9 @@ impl CompressionStrategy for DefaultCompressionStrategy {
         field: &Field,
         data: &DataBlock,
     ) -> Result<(Box<dyn BlockCompressor>, CompressiveEncoding)> {
-        match data {
+        let field_params = Self::parse_field_metadata(field);
```

Comment:
We should probably be following the same pattern we follow in miniblock (in fact, this should probably be a helper function we use in all thread `create_...` functions:

```
        let mut field_params = self
            .params
            .get_field_params(&field.name, &field.data_type());
        // Override with field metadata if present (highest priority)
        let metadata_params = Self::parse_field_metadata(field);
        field_params.merge(&metadata_params);
```